### DISCUSSION TITLE
how to increase airflow scheduler task running concurrency

### DISCUSSION THREAD

**Discussion by xingyang-007** [apache/airflow#31734] [Q&A]
we have many dags and each dag has about >10 tasks,
and we use kubernetes executor to make dynamic worker,
the current problem is total task number is huge but concurrent running task is very few(mostly 1-3),
does scheduler has any configuration to increase the concurrent running task num ?
we have tried "parallelism" but no effect.

**Comment by potiuk** [comment]
Look at the docs: 
* https://airflow.apache.org/docs/apache-airflow/stable/faq.html#faq
* https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/scheduler.html#fine-tuning-your-scheduler-performance

Maybe they wil help you to determine the roo cause. Also look at resource usages of your K8S cluster. Airlfow will not manage resources on k8s - this is k8s tasks, if for some reasons your tasks are using all the resources (memory/cpu etc), k8s will run as many of them as many resources it has - but this is beyond Airflow.

**Reply by xingyang-007** [reply]
i found that scheduler can be scale up to multi in document. and that works, the task queue time shorten a lot now.
thanks a lot.
